File: src/main/java/com/dh/ondot/member/api/response/HomeAddressResponse.java
package com.dh.ondot.member.api.response;

import com.dh.ondot.member.domain.Address;

public record HomeAddressResponse(
        String roadAddress,
        Double longitude,
        Double latitude
) {
    public static HomeAddressResponse from(Address address) {
        return new HomeAddressResponse(
                address.getRoadAddress(),
                address.getLongitude(),
                address.getLatitude()
        );
    }
}

File: src/main/kotlin/com/dh/ondot/member/presentation/response/HomeAddressResponse.kt
package com.dh.ondot.member.presentation.response

import com.dh.ondot.member.domain.Address

data class HomeAddressResponse(
    val roadAddress: String,
    val longitude: Double,
    val latitude: Double,
) {
    companion object {
        fun from(address: Address): HomeAddressResponse =
            HomeAddressResponse(
                roadAddress = address.roadAddress,
                longitude = address.longitude,
                latitude = address.latitude,
            )
    }
}
